pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -262,6 +263,84 @@ func (b *OFBridge) IsConnected() bool {
 	return b.ofSwitch.IsReady()
 }
 
+func (b *OFBridge) AddFlowsInBundle(addflows []Flow, modFlows []Flow, delFlows []Flow) error {
+	// Create a new transaction.
+	tx := b.ofSwitch.NewTransaction(ofctrl.Atomic)
+	// Open a bundle on the OFSwitch.
+	if err := tx.Begin(); err != nil {
+		return err
+	}
+
+	syncFlows := func(flows []Flow, operation int) error {
+		for _, flow := range flows {
+			ofFlow := flow.(*ofFlow)
+			ofFlow.Flow.NextElem = ofFlow.lastAction
+			// "AddFlow" operation is async, the function returns error when constructing and sending BundleAdd message.
+			// It doesn't represent all Openflow entries are added into the bundle even if the function
+			// returns nil. The number of the Openflow entries successfully added into the bundle is returned from function "Complete".
+			flowMod, err := ofFlow.Flow.GenerateFlowModMessage(operation)
+			if err != nil {
+				return err
+			}
+			if err := tx.AddFlow(flowMod); err != nil {
+				// Close the bundle and abort it if there is error when adding the FlowMod message.
+				_, err := tx.Complete()
+				if err == nil {
+					tx.Abort()
+				}
+				return err
+			}
+		}
+		return nil
+	}
+
+	// Install new Openflow entries with the opened bundle.
+	if err := syncFlows(addflows, openflow13.FC_ADD); err != nil {
```

Comment:
I saw addFlows, modFlows, delFlows could be nil? Then you should check them first.

Reply:
If we do not really have a case of multiple operations in the same transaction, maybe consider to make the operation an argument.

Reply:
I will add check for the parameters.

Network Policy might have multiple operations case, especially when calling `AddPolicyRuleAddress` or `DelPolicyRuleAddress`.